server/consumer.go

```diff
@@ -374,6 +380,24 @@ func (mset *Stream) AddConsumer(config *ConsumerConfig) (*Consumer, error) {
 		}
 	}
 
+	// Check if we have a rate limit set.
+	if config.RateLimit != 0 {
+		// TODO(dlc) - Make sane values or error if not sane?
```

Comment:
Is choosing bits per sec contributing to this? I understand that bandwidth, etc.. is in bits per sec, but at the higher level that is NATS people talk about msgs or bytes per second, not bit. Just a thought.

Reply:
I thought about that but think that if they were to limit it would be based on constraints expressed in bps.

Reply:
I think the bps is a bit weird - both in concept and implementation.

Generally when someone says 512 bps in terms of network pipes the TCP stack will tend to continuously send that many bits. Easily done when packets can be fragmented and so forth.

With the way this work if I set it to something low like 512 and when my messages are a few KB big for some reason I get a huge burst of 300+ messages (no idea why) but then I will only get one message per minute which is a bit weird.

With more realistic numbers it works a bit better but you can still see just pauses while it waits for enough seconds to pass to fit a message in.

This is all fine and as expected given the implementation and that we do not support message fragmentation etc (other than the 300 burst seems weird).  But it demonstrates that the bps concept against message orientated world is a bit orthogonal and yields a really weird outcome

Reply:
obviously setting this to something stupid like 512 is a mistake but I wanted to see how it works :)

Reply:
How does it do at 300k or so, like a 3G speed?

Reply:
it'll appear smoother for sure, but big messages cause hicups, I'll also try some more to reproduce this huge burst I saw